src/webcomponents/commons/tool-settings-update.js

```diff
@@ -0,0 +1,285 @@
+        return {
+            study: {
+                type: Object,
+            },
+            toolSettings: {
+                type: Object,
+            },
+            locus: {
+                type: Object,
+            },
+            opencgaSession: {
+                type: Object,
+            },
+        };
+    }
+
+    // --- PRIVATE METHODS ---
+
+    #init() {
+        this.toolSettings = {};
+        this.isLoading = false;
+        this.DEFAULT_TOOLPARAMS = {};
+        this._toolSettings = {}; // Local copy for tool settings
+        this._listStudies = [];
+    }
+
+    #setLoading(value) {
+        this.isLoading = value;
+        this.requestUpdate();
+    }
+
+    // TODO 20230210 Vero:
+    //  1. Move to utils-new.js.
+    //  2. Repeated in opencga-update
+    //  3. Rename first parameter and function name
+    #getResourceName(name, type) {
+        const words = name
+            .toLowerCase()
+            .split("_");
+
+        switch (type) {
+            case "event":
+                return words
+                    .reduce((result, word) => result + UtilsNew.capitalize(word));
+            case "label":
+                return words
+                    .map(word => UtilsNew.capitalize(word))
+                    .join(" ");
+        }
+    }
+
+    // --- UPDATE ---
+
+    update(changedProperties) {
+        if (changedProperties.has("study") || changedProperties.has("toolSettings")) {
+            this.componentObserver();
+        }
+        super.update(changedProperties);
+    }
+
+    #initOriginalObjects() {
+        // The original settings and study are maintained. A copy is used for previewing the ongoing changes in json
+        this._study = UtilsNew.objectClone(this.study);
+        this._toolSettings = UtilsNew.objectClone(this.toolSettings);
+        this.updatedFields = {};
+        this._config = {
+            ...this.getDefaultConfig(),
+            ...this.config,
+        };
+    }
+    // --- OBSERVERS ---
+    componentObserver() {
+        if (this.study && this.toolSettings && this.opencgaSession) {
+            this.#initOriginalObjects();
+        }
+    }
+
+    // --- EVENTS ---
+    onFieldChange(e) {
+        // The json has been modified, so we need to:
+        // FIXME: switch case
+        // 1. To update the local object settings where we store the settings modifications
+        if (e.detail.value?.json) {
+            this._toolSettings = {...e.detail.value?.json};
+        }
+        if (e.detail.param === "id") {
+            this._listStudies = e.detail.value.split(",");
+        }
+        // if (e.detail.param === "default") {debugger;}
+        // To notify that the json has been modified
+        // CAUTION 20230208 Vero: the toolSettings json has been updated
+        LitUtils.dispatchCustomEvent(this,
+            "studyToolSettingsUpdate",
+            null,
+            {_toolSettings: this._toolSettings}
+        );
+        this.requestUpdate();
```

Comment:
Please reformat this line, as some of the arguments can be placed in the same line:

```js
LitUtils.dispatchCustomEvent(this, "studyToolSettingsUpdate", null, {
    _toolSettings: this._toolSettings,
});
```

Also, I think that you may rename the `_toolSettings` para of the event as `toolSettings`.